src/packages/core/src/servers/http-server.ts

```diff
@@ -237,13 +243,18 @@ export default class HttpServer {
     // OPTIONS responses don't have a body, so respond with `204 No Content`...
     sendResponse(
       response,
+      this.#isClosing,
       HttpResponseCodes.NO_CONTENT,
       null,
       null,
       writeHeaders
     );
   };
+
   public close() {
-    // currently a no op.
+    // flags the server as closing, indicating the connection should be closed with subsequent responses
+    // as there is no way presently to close existing connections outside of the request/response context
+    // see discussion: https://github.com/uNetworking/uWebSockets.js/issues/663#issuecomment-1026283415
+    this.#isClosing = true;
   }
-}
+}
```

Comment:
Let's keep the newline at the end of files. Some devs like to use `cat` to view files, and the newline makes that process easier (so I've been told). :-)

```suggestion
}

```